elastic/kibana
# [Synthetics] Sorting and infinite scroll

## Summary

Relates to https://github.com/elastic/synthetics-dev/issues/101
Resolves https://github.com/elastic/kibana/issues/135158

Adds sorting and infinite scroll to Overview monitor management page

<img width="1516" alt="Screen Shot 2022-10-03 at 8 35 10 AM" src="https://user-images.githubusercontent.com/11356435/193578475-12e55a9b-b2db-46e2-a0b0-f8e8ec2e4315.png">

### Testing
1. The easiest way to test many monitors is to use project monitors. 
2. Create a synthetics testing directory by using `npx @elastic/synthetics init`, or use an existing directory by chaning the url in the `synthetics.config.ts` file.
3. Wrap a journey in an for loop so that you can create a variable amount of monitors. For example
```

for (let i = 0; i < 100; i++) {
  journey(`addition and completion of single task been 2 ${i}`, ({ page, params }) => {
    const testText = "Don't put salt in your eyes";

    loadAppStep(page, params.url);
    addTaskStep(page, testText);
    checkForTaskStep(page, testText);
    destroyTaskStep(page, testText);
  });

  journey(`adding and removing multiple tasks been ${i}`, ({ page, params }) => {
    const testTasks = ['Task 1', 'Task 2', 'Task 3'];

    loadAppStep(page, params.url);
    testTasks.forEach(t => {
      addTaskStep(page, t);
    });

    // remove the middle task and check that it worked
    destroyTaskStep(page, testTasks[1]);

    // add a new task and check it exists
    addTaskStep(page, 'Task 4');
  });
}
```
4. Connect to the service via Kibana with the following config. Or if you prefer, connect to your local synthetics service environment https://p.elstc.co/paste/ZJDStaem#n1UAnhWT9ali+T1fxqTNQN+WLqgB469VJzk7nZyUK4J
4. If you're connected to the dev service environment via the above config, you can define `us_central` as your location in your synthetics.config.ts file.
```

export default env => {
  const config: SyntheticsConfig = {
    params: {
      url: 'https://elastic.github.io/synthetics-demo/',
    },
    playwrightOptions: {
      ignoreHTTPSErrors: true,
    },
    /**
     * Configure global monitor settings
     */
    monitor: {
      schedule: 5,
      locations: ['us_central'],
      privateLocations: [],
    },
    /**
     * Project monitors settings
     */
    project: {
      id: 'test2',
      url: 'http://localhost:5601/dcu',
      space: 'default',
    },
  };
  if (env !== 'development') {
    /**
     * Override configuration specific to environment
     * Ex: config.params.url = ""
     */
  }
  return config;
};
```
6. Push the monitors by creating an API key and running `SYNTHETICS_API_KEY={theykey} npm run push`.
7. You should now have any many monitors as you wish to test with. Ideally, you should create about 100 or 200 to test infinite scrolling. Monitors should load as you scroll down the page
8. When interacting with the sort dropdown, monitors should be sorted correctly according to the specifications, either alphabetically, by status, or by last modified.

### Checklist

Delete any items that are not applicable to this PR.
